frontend/src/concepts/pipelines/content/pipelineSelector/PipelineVersionSelector.tsx

```diff
@@ -51,6 +52,12 @@ const PipelineVersionSelector: React.FC<PipelineVersionSelectorProps> = ({
     data: versions,
   } = usePipelineVersionSelector(pipelineId);
 
+  const supportedVersions = React.useMemo(
```

Comment:
this is fine to do for the create page, but the pipeline version selector on the details page can have the invalid version selected but it is not visible in the dropdown. you might need to add a condition to the version selector for if it should filter out invalid versions

Reply:
Made that change, now i'm stuck with other error.

https://github.com/user-attachments/assets/13845e9a-76ed-4b12-acd5-7148cbe4b3aa

When changing the pipeline version, we see this error screen. The issue originates from [this](https://github.com/opendatahub-io/odh-dashboard/blob/f2ce42bedb9a0badd58ef8ec1774a00166a88d05/frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetails.tsx#L262) line, where the code is accessing the `name` property from the pipeline spec.

This error might be due to an improper state update of the `topologyError` variable. I've tried debugging but haven't found a solution yet. Your input would be appreciated.

Reply:
ill pull the changes and see what i can do

Reply:
it think its getting too complicated with he topology hook sending back an error, just test if the version is invalid from the get go and use that instead of the topology error. you can probably revert the topoloigy hook to just catching the error and not retuning a bunch of error and loaded states

Here is a simple diff
```diff
diff --git a/frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetails.tsx b/frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetails.tsx
index d5e0ceeb..dd155052 100644
--- a/frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetails.tsx
+++ b/frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetails.tsx
@@ -54,24 +54,18 @@ const PipelineDetails: PipelineCoreDetailsPageComponent = ({ breadcrumbPath }) =
     usePipelineVersionById(pipelineId, pipelineVersionId);
   const [pipeline, isPipelineLoaded, pipelineLoadError] = usePipelineById(pipelineId);
 
-  const {
-    nodes,
-    loaded: isTopologyLoaded,
-    error: topologyError,
-  } = usePipelineTaskTopology(pipelineVersion?.pipeline_spec);
+  const { nodes } = usePipelineTaskTopology(pipelineVersion?.pipeline_spec);
+
+  const isInvalidPipelineVersion = isArgoWorkflow(pipelineVersion?.pipeline_spec);
 
   const selectedNode = React.useMemo(() => {
-    if (topologyError) {
+    if (isInvalidPipelineVersion) {
       return null;
     }
     return nodes.find((n) => n.id === selectedId);
-  }, [topologyError, selectedId, nodes]);
+  }, [isInvalidPipelineVersion, selectedId, nodes]);
 
-  const isLoaded =
-    isPipelineVersionLoaded &&
-    isPipelineLoaded &&
-    !!pipelineVersion?.pipeline_spec &&
-    isTopologyLoaded;
+  const isLoaded = isPipelineVersionLoaded && isPipelineLoaded && !!pipelineVersion?.pipeline_spec;
 
   if (pipelineVersionLoadError || pipelineLoadError) {
     const title = 'Pipeline version not found';
@@ -171,7 +165,7 @@ const PipelineDetails: PipelineCoreDetailsPageComponent = ({ breadcrumbPath }) =
           )
         }
       >
-        {topologyError ? (
+        {isInvalidPipelineVersion ? (
           <PipelineNotSupported />
         ) : (
           <PageSection
```

Reply:
thank you it's working fine now, reverted back the changes in usePipelineTaskTopology